### PR TITLE
change to scripts installation instead of manual deps and flags and code improvement in general

### PR DESCRIPTION
**Pull request by ashpect**
Changes Summery :

1. Changed structure to having 3 subcommands dev, deps and runtime for 3 kind of images.
2. For flags, made a new file flags.go to store all flags and had functions written to set flags. 
3. Exported struct and making variable when needed.
4. Also created dep/dev options struct which get passed down the cli func, is way better if we want to add ui cli libraries later.
5. Changed manual work to running metacall scripts.

Example Usage : 
`"builder runtime -b develop --cli nodejs typescript go rust wasm java c cobol"`, (will create a runtime image with metacall cli installed)

### CODE REVIEW

**Comment by viferga** [comment]
@ashpect can you resolve the conflicts? apparently cmd/builder/dev.go has some.

**Comment by ashpect** [comment]
@viferga resolved

**Comment by ashpect** [comment]
@viferga Pls review whenever you get time.

File: pkg/staging/deps.go [hunk 1]
```diff
@@ -1,31 +1,141 @@
...
+		MetaCallClone(branch).
+		MetacallRuntime(cmdArgs).
+		Root()
+}
+
+func AddCli(src llb.State, dst llb.State) llb.State {
+	return dst.With(copyFrom(src, "/usr/local/bin/metacallcli*", "/usr/local/bin/metacall"))
+}
+
+func CopyFromBuilder(src llb.State, dst llb.State) llb.State {
```
Comment (viferga) [inline pkg/staging/deps.go:107]: This MUST be avoided, we have to use diff for properly implement this in a generic way, this is the main purpose of using buildkit... and this defeats that purpose.

Comment (ashpect) [reply pkg/staging/deps.go]: Resolved in the latest commits. Changes can be seen in runtime.go and deps.go.

**Comment by ashpect** [comment]
Screenshot 2024-06-01 at 2 06 15 AM
Screenshot 2024-06-01 at 2 06 51 AM

**Pull request merged**